src/Lykke.RabbitMq.Azure/MessagePackBlobPublishingQueueRepository.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using AzureStorage;
+using Lykke.RabbitMqBroker.Publisher;
+using MessagePack;
+
+namespace Lykke.RabbitMq.Azure
+{
+    public sealed class MessagePackBlobPublishingQueueRepository : IPublishingQueueRepository
+    {
+        private const string Container = "RabbitMqPublisherMessages-v-3-1-mp";
+
+        private readonly IBlobStorage _storage;
+        private readonly string _instanceName;
+
+        /// <param name="storage"></param>
+        /// <param name="instanceName">Instance name, required when multiple publishers publishes to single exchange</param>
+        public MessagePackBlobPublishingQueueRepository(IBlobStorage storage, string instanceName = null)
+        {
+            _storage = storage;
+            _instanceName = instanceName;
+        }
+
+        /// <summary>
+        /// Saves messages to azure storage
+        /// </summary>
+        /// <param name="items">Serialized messages</param>
+        /// <param name="exchangeName">Exchange name</param>
+        /// <returns></returns>
+        public async Task SaveAsync(IReadOnlyCollection<byte[]> items, string exchangeName)
+        {
+            using (var stream = new MemoryStream())
+            {
+                MessagePackSerializer.Serialize(stream, items);
+
+                await stream.FlushAsync();
+                stream.Seek(0, SeekOrigin.Begin);
+
+                await _storage.SaveBlobAsync(Container, GetKey(exchangeName), stream);
+            }
+        }
+
+
+        public async Task<IReadOnlyCollection<byte[]>> LoadAsync(string exchangeName)
+        {
+            if (!await _storage.HasBlobAsync(Container, GetKey(exchangeName)))
+            {
+                return null;
+            }
```

Comment:
I think it is better return an empty array to avoid NPE

Reply:
caller should has opportunity to distinct empty result and the missing blob. null - is the missing blob.